api/db/populate-notifications.js

```diff
@@ -0,0 +1,222 @@
+};
+
+const createNotifications = async (verbose) => {
+  console.log('\n', '\x1b[0m', 'Populating notifications collection...');
+  displayLogs = verbose;
+
+  if ((await Notification.countDocuments()) !== 0) {
+    const proceed = await createPrompt(
+      'The notifications collection already exists. If you continue, the existing collection will be dropped to avoid duplication. Do you want to continue?'
+    );
+
+    if (proceed) {
+      await Notification.collection.drop();
+      console.log('\x1b[31m', 'Dropped old notifications collection');
+    } else {
+      console.log('\x1b[33m', 'Aborted notifications collection population');
+      return;
+    }
+  }
+
+  const notificationsArray = [];
+  const usersList = await User.find();
+  const eventsList = await Event.find();
+  const attendantsList = await Attendant.find();
+  const commentsList = await Comment.find();
+  const friendshipsList = await Friendship.find();
+  const messagesList = await Message.find();
+
+  // For each user
+  for (const user of usersList) {
+    let read = user.emailConfirmed;
+    // Email confirmation notification
+    notificationsArray.push(
+      await createNotification({
+        user,
+        type: types[0],
+        read,
+      })
+    );
+    // Create event notification
+    read = eventsList.some(
+      (event) => event.createdBy.toString === user._id.toString()
+    );
+    notificationsArray.push(
+      await createNotification({
+        user,
+        type: types[1],
+        read,
+      })
+    );
+    // Create upload avatar notification
+    read = user.avatar ? true : false;
+    notificationsArray.push(
+      await createNotification({
+        user,
+        type: types[2],
+        read,
+      })
+    );
+    // Selected interests notification
+    const interests = await Interest.find({ user });
+    read = interests.length > 0;
+    notificationsArray.push(
+      await createNotification({
+        user,
+        type: types[3],
+        read,
+      })
+    );
+
+    // Create attendee and comment notifications
+    const usersEvents = eventsList.filter(
+      (event) => event.createdBy.toString() === user._id.toString()
+    );
+    for (const event of usersEvents) {
+      const attendees = attendantsList.filter(
+        (attendee) =>
+          attendee.event.toString() === event._id.toString() &&
+          attendee.user.toString() !== user._id.toString()
+      );
+      for (const attendee of attendees) {
+        read = Math.random() > 0.5;
+        const notificationParams = {
+          user,
+          type: types[4],
+          read,
+          referencedUser: attendee.user,
+          referencedEvent: event,
+        };
+        notificationsArray.push(await createNotification(notificationParams));
+      }
+
+      const comments = commentsList.filter(
+        (comment) =>
+          comment.event.toString() === event._id.toString() &&
+          comment.user.toString() !== user._id.toString()
+      );
```

Comment:
This will only generate the new comment notification for the creator of the event, no? It should be generated for all the event attendees to replicate the real application behavior